genesyscloud/outbound_filespecificationtemplate/genesyscloud_outbound_filespecificationtemplate_proxy.go

```diff
@@ -0,0 +1,273 @@
+	name string,
+) (id string, retryable bool, err error) {
+	fileSpecificationTemplates, _, err := p.outboundApi.GetOutboundFilespecificationtemplates(
+		100, 1, true, "", name, "", "")
+	if err != nil {
+		return "", false, err
+	}
+
+	if fileSpecificationTemplates.Entities == nil || len(*fileSpecificationTemplates.Entities) == 0 {
+		return "", true, fmt.Errorf("No outbound filespecificationtemplate with name %s", name)
+	}
+
+	for _, fileSpecificationTemplate := range *fileSpecificationTemplates.Entities {
+		if *fileSpecificationTemplate.Name == name {
+			log.Printf("Retrieved the outbound file specification template id %s by name %s",
+				*fileSpecificationTemplate.Id, name)
+			return *fileSpecificationTemplate.Id, false, nil
+		}
+	}
+
+	return "", true, fmt.Errorf("Unable to find outbound file specification template with name %s", name)
+}
+
+// getOutboundFilespecificationtemplateByIdFn is an implementation of the function
+// to get a Genesys Cloud outbound filespecificationtemplate by id
+func getOutboundFilespecificationtemplateByIdFn(
+	ctx context.Context,
+	p *outboundFilespecificationtemplateProxy,
+	id string,
+) (outboundFilespecificationtemplate *platformclientv2.Filespecificationtemplate, statusCode int, err error) {
+	fst, resp, err := p.outboundApi.GetOutboundFilespecificationtemplate(id)
+	if err != nil {
+		return nil, resp.StatusCode, fmt.Errorf(
+			"Failed to retrieve file specification template by id %s: %s", id, err)
+	}
+
+	return fst, resp.StatusCode, nil
+}
+
+// updateOutboundFilespecificationtemplateFn is an implementation of the function
+// to update a Genesys Cloud outbound filespecificationtemplate
+func updateOutboundFilespecificationtemplateFn(
+	ctx context.Context,
+	p *outboundFilespecificationtemplateProxy,
+	id string,
+	outboundFilespecificationtemplate *platformclientv2.Filespecificationtemplate,
+) (*platformclientv2.Filespecificationtemplate, error) {
+	fst, _, err := getOutboundFilespecificationtemplateByIdFn(ctx, p, id)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to file specification template by id %s: %s", id, err)
+	}
+
+	outboundFilespecificationtemplate.Version = fst.Version
+	fileSpecificationTemplate, _, err := p.outboundApi.PutOutboundFilespecificationtemplate(id, *outboundFilespecificationtemplate)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to update file specification template: %s", err)
+	}
+	return fileSpecificationTemplate, nil
+}
+
+// deleteOutboundFilespecificationtemplateFn is an implementation function for
+// deleting a Genesys Cloud outbound filespecificationtemplate
+func deleteOutboundFilespecificationtemplateFn(
+	ctx context.Context,
+	p *outboundFilespecificationtemplateProxy,
+	id string,
+) (response *platformclientv2.APIResponse, err error) {
+	resp, err := p.outboundApi.DeleteOutboundFilespecificationtemplate(id)
+	if err != nil {
+		return resp, fmt.Errorf("Failed to delete file specification template: %s", err)
+	}
+
+	return resp, nil
+}
```

Comment:
Very minor comment - this could be one line
`return p.outboundApi.DeleteOutboundFilespecificationtemplate(id)`